### PR TITLE
add fmt/7.0.0 + if header only: copy header files only and properly define import target

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **fmt/7.0.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by prince-chrismc** [comment]
Damn u r fast!!!!

File: recipes/fmt/all/conanfile.py [hunk 1]
```diff
@@ -82,11 +79,13 @@ def package_id(self):
             del self.info.options.with_fmt_alias
 
     def package_info(self):
-        if self.options.with_fmt_alias:
-            self.cpp_info.defines.append("FMT_STRING_ALIAS=1")
         if self.options.header_only:
-            self.cpp_info.defines.append("FMT_HEADER_ONLY")
+            self.cpp_info.components["fmt-header-only"].defines.append("FMT_HEADER_ONLY=1")
```
Comment (prince-chrismc) [inline recipes/fmt/all/conanfile.py:83]: Nice catch, https://github.com/fmtlib/fmt/blob/b287c37c651d54669032e3e4685b4423409e4d00/CMakeLists.txt#L240

**Comment by uilianries** [comment]
@SpaceIm I think the error is related to minimal compiler version supported

**Comment by SpaceIm** [comment]
@ulirianries I've open an issue: https://github.com/fmtlib/fmt/issues/1756

I tried with Visual Studio 2019 and CXX_STANDARD 11, 14 or 17 in test_package: undefined references whatever the value.

**Comment by prince-chrismc** [comment]
Now we hit the other issue!

**Comment by prince-chrismc** [comment]
https://github.com/fmtlib/fmt/commit/fbf3b943cc9a515d76956d8e766455201f11fd7d

**Comment by uilianries** [comment]
Re-manage test_package to use `cmake_find_package` instead, it will fix this error.

**Comment by SpaceIm** [comment]
Maybe a better idea to drop 7.0.0 and only package 7.0.1: https://github.com/fmtlib/fmt/releases/tag/7.0.1

**Comment by prince-chrismc** [comment]
> Maybe a better idea to drop 7.0.0 and only package 7.0.1: https://github.com/fmtlib/fmt/releases/tag/7.0.1

:100:  not point having a broken release

**Comment by SSE4** [comment]
CI stuck?

**Comment by uilianries** [comment]
Yes! Second time, same result. All builds Okay, but CI status is pending yet.

/cc @jgsogo @danimtb

**Comment by jgsogo** [comment]
In the CI everything is OK, but it says:

> Could not update commit status, please check if your scan credentials belong to a member of the organization or a collaborator of the repository and repo:status scope is selected

I'm merging this one, everything is OK, only this last step is failing 🤷

**Pull request merged**